packages/smooth_app/lib/pages/prices/price_add_product_card.dart

```diff
@@ -96,39 +107,71 @@ class _PriceAddProductCardState extends State<PriceAddProductCard> {
       context,
       listen: false,
     );
-    for (int i = 0; i < priceModel.length; i++) {
-      final PriceAmountModel model = priceModel.elementAt(i);
-      if (model.product.barcode == barcode) {
-        await showDialog<void>(
-          context: context,
-          builder: (final BuildContext context) => SmoothAlertDialog(
-            body: Text(appLocalizations.prices_barcode_already(barcode)),
-            positiveAction: SmoothActionButton(
-              text: appLocalizations.okay,
-              onPressed: () => Navigator.of(context).pop(),
-            ),
+
+    bool barcodeAlreadyThere(final String barcode) {
+      for (int i = 0; i < priceModel.length; i++) {
+        final PriceAmountModel model = priceModel.elementAt(i);
+        if (model.product.barcode == barcode) {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    final List<String> alreadyThere = <String>[];
```

Comment:
I think you can use a `Set` directly and thus remove `barcodeAlreadyThere`, by calling only the `Set.contain` method.

Reply:
> I think you can use a `Set` directly and thus remove `barcodeAlreadyThere`, by calling only the `Set.contain` method.

Indeed, but I assumed the order was important too for the user, and Set is unordered.